extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxMDC.java

```diff
@@ -314,7 +314,7 @@ private Map<String, Object> contextualDataMap(Context ctx) {
         }
 
         ConcurrentMap<Object, Object> lcd = Objects.requireNonNull((ContextInternal) ctx).localContextData();
-        return (ConcurrentMap<String, Object>) lcd.computeIfAbsent(VertxMDC.class.getName(),
+        return (ConcurrentMap<String, Object>) lcd.computeIfAbsent(VertxMDC.class.getName() + ctx.hashCode(),
```

Comment:
I don't believe this is correct.
For sure, it's incorrect when using nested context as you lose your MDC.